torch/distributions/geometric.py

```diff
@@ -88,7 +88,8 @@ def sample(self, sample_shape=torch.Size()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        value, probs = broadcast_all(value, self.probs.clone(memory_format=torch.contiguous_format))
+        value, probs = broadcast_all(value, self.probs)
+        probs = probs.clone(memory_format=torch.contiguous_format)
         probs[(probs == 1) & (value == 0)] = 0
```

Comment:
>My impression was that broadcasted tensors are routinely used in distributions, so requiring non-self-overlapping outputs could lead to significant memory penaly.

This is a bug fix. If `probs` is broadcasted here, then this line can write to locations where `value != 0` if it happens to overlap with a location where `value` is zero. The full tensor size will be required in the following calculation anyway, so I think this is fine.

The same reasoning applies to the change in multinomial.